src/Language/PureScript/TypeChecker/Entailment/Coercible.hs

```diff
@@ -531,7 +531,7 @@ insoluble k a b =
   -- "Consider adding a type annotation" hint, because annotating kinds to
   -- instantiate unknowns in Coercible constraints should never resolve
   -- NoInstanceFound errors.
-  errorMessage $ NoInstanceFound (srcConstraint Prim.Coercible [k] [a, b] Nothing) [] (any containsUnknowns [a, b])
+  errorMessage $ NoInstanceFound (srcConstraint Prim.Coercible [k] [a, b] Nothing) [] (if any containsUnknowns [a, b] then Unknowns else NoUnknowns)
```

Comment:
This was a 'quick change', but it needs to be thought out more.

Reply:
I'm using [CoercibleHigherKindedData.](https://github.com/purescript/purescript/blob/master/tests/purs/failing/CoercibleHigherKindedData.purs) and its [test output](https://github.com/purescript/purescript/blob/master/tests/purs/failing/CoercibleHigherKindedData.out) to guide my thought process here.

This particular test fails to compile because of the type role `representational` on the `Proxy` type's `a` type variable. Because `Binary a` is not a newtype of `Unary` or vice versa, this particular example fails to compile. Moreover, the [`a` and `b` values](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/TypeChecker/Entailment.hs#L452C14-L453) passed to [`insoluble`](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/TypeChecker/Entailment/Coercible.hs#L524-L534) are not the original values used when solving the class constraint as can be seen by the test's golden output's error message.

So, even if we took a similar approach in other tests' error messages (i.e. producing the list of possible VTA expressions one can use to resolve the unknowns), using VTAs for `unaryToBinary` would not fix the problem.

Perhaps `Coercible` is a special case where listing the VTAs one could use in the error message should not be done and instead documentation on `coerce` itself would be better?